oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -97,11 +97,9 @@
                 WorkflowJob.Status.FAILED);
     private static final List<WorkflowJob.Status> WF_SUSPEND_PRECOND = Arrays.asList(WorkflowJob.Status.RUNNING);
     private static final List<WorkflowJob.Status> WF_RESUME_PRECOND = Arrays.asList(WorkflowJob.Status.SUSPENDED);
-    private static final List<WorkflowJob.Status> WF_RERUN_PRECOND =
-        Arrays.asList(WorkflowJob.Status.FAILED, WorkflowJob.Status.KILLED, WorkflowJob.Status.SUCCEEDED);
     private static final List<CoordinatorAction.Status> COORD_RERUN_PRECOND =
-        Arrays.asList(CoordinatorAction.Status.TIMEDOUT, CoordinatorAction.Status.FAILED);
-
+        Arrays.asList(CoordinatorAction.Status.TIMEDOUT, CoordinatorAction.Status.FAILED,
+                CoordinatorAction.Status.KILLED, CoordinatorAction.Status.SUCCEEDED);
```

Comment:
Checked. SUSPENDED actions don't get rerun. Rerun coordinator action must be in TIMEDOUT/SUCCEEDED/KILLED/FAILED. 

"Rerun a SUSPENDED coordinator job, the status will reset to RUNNING." - Means even if a coord job is suspended, when some actions are rerun, the job (and not action) moves to RUNNING. I saw this happen. Same happens to DONEWITHERROR (moves to RUNNING when any action is rerun)

So, won't add SUSPENDED.